src/main/java/com/gradle/CustomBuildScanEnhancements.java

```diff
@@ -395,21 +400,22 @@ public void execute(BuildScanExtension buildScan) {
                 buildScan.value("Git status", gitStatus);
             }
 
-            if (isNotEmpty(gitRepo) && isNotEmpty(gitCommitId)) {
-                if (gitRepo.contains("github.com/") || gitRepo.contains("github.com:")) {
-                    Matcher matcher = Pattern.compile("(.*)github\\.com[/|:](.*)").matcher(gitRepo);
-                    if (matcher.matches()) {
-                        String rawRepoPath = matcher.group(2);
-                        String repoPath = rawRepoPath.endsWith(".git") ? rawRepoPath.substring(0, rawRepoPath.length() - 4) : rawRepoPath;
-                        buildScan.link("Github source", "https://github.com/" + repoPath + "/tree/" + gitCommitId);
-                    }
-                } else if (gitRepo.contains("gitlab.com/") || gitRepo.contains("gitlab.com:")) {
-                    Matcher matcher = Pattern.compile("(.*)gitlab\\.com[/|:](.*)").matcher(gitRepo);
-                    if (matcher.matches()) {
-                        String rawRepoPath = matcher.group(2);
-                        String repoPath = rawRepoPath.endsWith(".git") ? rawRepoPath.substring(0, rawRepoPath.length() - 4) : rawRepoPath;
-                        buildScan.link("GitLab Source", "https://gitlab.com/" + repoPath + "/-/commit/" + gitCommitId);
+            Optional<String> gitHubUrl = envVariable("GITHUB_SERVER_URL", providers);
+            Optional<String> gitHubRepository = envVariable("GITHUB_REPOSITORY", providers);
+            if (gitHubUrl.isPresent() && gitHubRepository.isPresent()) {
```

Comment:
Why don't you check for an empty gitCommitId here?

Reply:
I didn't really think that was a possible scenario, I was thinking maybe just in the case of a freshly initialized repository without commits? But even then it returns an error message, so I'm not sure when it would be empty. So we could potentially check that it is a valid hash?

Reply:
You check for an empty commit id in the block below.

Reply:
Done.